src/H5.c

```diff
@@ -1226,3 +1233,63 @@ H5is_library_terminating(bool *is_terminating /*out*/)
 
     FUNC_LEAVE_API_NOINIT(ret_value)
 } /* end H5is_library_terminating() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5_user_cb_prepare
```

Comment:
Save the state of the library when callback leaves the library.  Includes error stack and threadsafe reentry state currently